apps/site/lib/site_web/channels/predictions_channel.ex

```diff
@@ -21,7 +21,11 @@ defmodule SiteWeb.PredictionsChannel do
     [route_id, stop_id, direction_id] = String.split(route_stop_direction, ":")
     predictions = predictions_subscribe_fn.(route_id, stop_id, direction_id)
 
-    {:ok, %{predictions: predictions}, socket}
+    # remove old predictions
+    future_predictions = Enum.filter(predictions, &is_in_future?/1)
+    # immediately push current predictions to socket
+    send(self(), {:new_predictions, future_predictions})
+    {:ok, socket}
```

Comment:
**question:** I'm a bit confused about this change. Returning `{:ok, %{predictions: predictions}, socket}` should already make the data available to the client as soon as it joins, though it's a little bit confusing because it's accessed in a different way from the usual results of a push. We actually had a bug in Skate caused by this issue but we were able to fix it client-side in mbta/skate#1575.

Reply:
That's from an older commit -- the current code removes the `future_predictions = Enum.filter(predictions, &is_in_future?/1)` from here and puts it in the `handle_info({:new_predictions, predictions}, socket)` clause, which is also handling some other data transformation. To clarify, we `send(self(), {:new_predictions, predictions})` so that the handle_info can do the data filtering before sending the data out. I hope this helps?

Reply:
Ahhhh, thanks for elaborating. That makes sense now. Have added a commit to make this change, great suggestion 👍🏼